client/my-sites/domains/domain-management/transfer/transfer-out/unlocked.jsx

```diff
@@ -62,81 +99,193 @@ class Unlocked extends React.Component {
 						break;
 				}
 				notices.error( errorMessage );
-
-				this.setState( { submitting: false } );
-			} else if ( hasPrivacyProtection ) {
-				notices.success( translate( 'We\'ve canceled your domain transfer. Your domain is now locked and ' +
-					'Privacy Protection has been enabled.' ) );
 			} else {
-				notices.success( translate( 'We\'ve canceled your domain transfer. Your domain is now locked back.' ) );
+				// Success.
+				this.setStateIfMounted( { sent: false } );
+
+				let successMessage;
+				if ( enablePrivacy && lockDomain ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer. Your domain is now re-locked and ' +
+						'Privacy Protection has been enabled.' );
+				} else if ( enablePrivacy ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer and ' +
+						'Privacy Protection has been re-enabled.' );
+				} else if ( lockDomain ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer and ' +
+						're-locked your domain.' );
+				} else {
+					successMessage = translate( 'We\'ve canceled your domain transfer. ' );
+				}
+
+				notices.success( successMessage );
 			}
 		} );
 	};
 
-	handleResendConfirmationCodeClick = () => {
-		this.setState( { submitting: true } );
+	isDomainAlwaysTransferrable() {
+		const {
+			domainLockingAvailable,
+			hasPrivacyProtection,
+		} = getSelectedDomain( this.props );
+		return ! domainLockingAvailable && ! hasPrivacyProtection;
+	}
+
+	renderCancelButton( domain ) {
+		const { pendingTransfer } = domain;
+
+		if ( ! pendingTransfer && this.isDomainAlwaysTransferrable() ) {
+			return null;
+		}
+
+		return ( <Button
+			onClick={ this.handleCancelTransferClick }
+			disabled={ this.state.submitting || ! this.state.sent }
+			compact>{ this.props.translate( 'Cancel Transfer' ) }
+		</Button> );
+	}
+
+	renderSendButton( domain ) {
+		const { translate } = this.props;
+		const { manualTransferRequired } = domain;
+
+		if ( manualTransferRequired && this.state.sent ) {
```

Comment:
Are we restricting manual transfers to sending the code once for a technical reason underlying those TLDs? I'm curious why no resends like other TLDs.